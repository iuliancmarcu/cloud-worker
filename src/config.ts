type Config = {
  debug: boolean;
  isSlave: boolean;
  redis: {
    host: string;
    port: number;
    password: string;
    tasksQueue: string;
    resultsChannel: string;
    resultsList: string;
  };
};

export function readEnv(key: string): string {
  const value = process.env[key];
  if (value == null) {
    throw new Error(`"${key}" not found in environment`);
  }

  return value;
}

let config: Config = {
  debug: process.env.DEBUG === 'true',
  isSlave: readEnv('TYPE') === 'slave',
  redis: {
    host: readEnv('REDIS_HOST'),
    port: parseInt(readEnv('REDIS_PORT')),
    password: readEnv('REDIS_PASSWORD'),
    tasksQueue: readEnv('REDIS_TASKS_QUEUE'),
    resultsChannel: readEnv('REDIS_RESULTS_CHANNEL'),
    resultsList: readEnv('REDIS_RESULTS_LIST'),
  },
};

export default config;
