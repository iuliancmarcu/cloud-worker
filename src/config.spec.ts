import { readEnv } from './config';

describe('config', () => {
  describe('readEnv', () => {
    it('throws an error if key is not found in env', () => {
      const key = 'FOOBAR_NO_CHANCE_TO_BE_IN_ENV';
      expect(() => readEnv(key)).toThrowErrorMatchingSnapshot();
    });
  });
});
