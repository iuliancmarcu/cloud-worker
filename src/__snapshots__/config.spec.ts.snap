// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config readEnv throws an error if key is not found in env 1`] = `"\\"FOOBAR_NO_CHANCE_TO_BE_IN_ENV\\" not found in environment"`;
